app/controllers/financials.js

```diff
@@ -0,0 +1,125 @@
+  EventEmitter.call(this);
+
+  this.url = url;
+  this.fileName = fileSystem.getFileName(this.url);
+  this.pathInCache = fileSystem.getPathInCache(this.url);
+  this.pathInDownload = fileSystem.getPathInDownload(this.url);
+  this.riseDisplayNetworkII = riseDisplayNetworkII;
+  this.logger = logger;
+};
+
+util.inherits(FinancialsController, EventEmitter);
+
+FinancialsController.prototype.getData = function(opts) {
+  let options = {};
+
+  if ( this.url ) {
+    options.url = this.url;
+
+    options.headers = {
+      "User-Agent": "request"
+    };
+    options.proxy = (this.riseDisplayNetworkII) ?
+      (this.riseDisplayNetworkII.get("activeproxy") || this.riseDisplayNetworkII.get("rcosproxy")) :
+      null;
+    options.timeout = config.requestTimeout;
+
+    if (opts) {
+      Object.assign(options.headers, opts);
+    }
+
+    fileSystem.getAvailableSpace(this.logger, (spaceInDisk)=>{
+      request(options, (error, res, body) => {
+        if (error || res.statusCode != 200) {
+          this.getSavedData((err, data) => {
+            if (err) { this.logger.error(err, null, this.url); }
+
+            if (data) {
+              this.emit("data", data);
+            } else {
+              if (error) {
+                this.emit("request-error", error);
+              } else {
+                this.emit("invalid-response", res.statusCode);
+              }
+            }
+          });
+
+        } else {
+          let fileSize = res.headers["content-length"];
+
+          fileSystem.isThereAvailableSpace(this.logger, (isThereAvailableSpace) => {
+            if(isThereAvailableSpace) {
+              fileSystem.addToDownloadTotalSize(fileSize);
+              this.saveData(body, fileSize);
+            } else {
+              this.logger.error("Insufficient disk space", fileSize, this.url);
+            }
+
+            this.emit("data", body);
+
+          }, spaceInDisk, fileSize);
+
+        }
+
+      });
+    });
+
+  }
+};
+
+FinancialsController.prototype.saveData = function(body, fileSize) {
+  fs.writeFile(this.pathInDownload, body, (err) => {
+    if (err) {
+      this.deleteFileFromDownload();
+      this.emit("file-error", err);
+    } else {
+      this.moveFileFromDownloadToCache();
+      fileSystem.removeFromDownloadTotalSize(fileSize);
+      this.emit("saved");
+    }
+  });
+};
+
+FinancialsController.prototype.getSavedData = function(cb) {
+  fileSystem.isCached(this.url, (cached) => {
+    if (!cached) { return cb(); }
+
+    fs.readFile(this.pathInCache, (err, data) => {
+      if (err) return cb(err);
+
+      return cb(null, data);
+    });
+  });
+};
+
+/* Move downloaded file form download folder to cache folder. */
+FinancialsController.prototype.moveFileFromDownloadToCache = function() {
+  fileSystem.move(this.pathInDownload, this.pathInCache, (err) =>{
+    if (err) this.emit("move-file-error", err);
+  });
+};
+
+FinancialsController.prototype.deleteFileFromDownload = function() {
```

Comment:
@stulees I would move this methods to fileSystem helper so reusing it on the file controller and here. Thoughts? cheers